src/main/helm/helm-release-manager.ts

```diff
@@ -108,7 +108,7 @@ export async function upgradeRelease(name: string, chart: string, values: any, n
 
     return {
       log: output,
-      release: getRelease(name, namespace, kubeconfigPath, kubectlPath),
+      release: await getRelease(name, namespace, kubeconfigPath, kubectlPath),
```

Comment:
Noticed no `catch` in this `try...finally`, will `await getRelease` throw? Or error is caught and handled somewhere else?

Reply:
Key difference here is that all back-end API responds go through same place which already catches fatal errors. That being said, if it fails fatally, API call is resolved with status code `422` and error message. And in this case, there is no non-fatal error to be handled. Goal is to get rid of try-catches in operational code for good (strategic places that handle errors for everything to allow error monitoring etc). Throwing is mechanism that is reserved for fatal errors which should prevent code to be executed and for non fatal errors we have better alternatives.